clio-client/src/main/scala/org/broadinstitute/clio/client/commands/ClioCommand.scala

```diff
@@ -33,7 +35,7 @@ sealed abstract class GetSchemaCommand[TI <: TransferIndex](val index: TI)
 sealed abstract class AddCommand[TI <: TransferIndex](val index: TI)
     extends ClioCommand {
   def key: index.KeyType
-  def metadataLocation: String
+  def metadataLocation: URI
```

Comment:
I chose `URI` instead of `Path` because:
1. Without the google-cloud-nio library, `Paths.get` of a string with a "gs" scheme throws a "no filesystem provider" exception
2. With the google-cloud-nio library, `Paths.get` of a string with a "gs" scheme eagerly tries to connect to the google bucket in the path, throwing an exception. On the client-side we don't want the connection to happen until after we've generated / parsed an access token, and on the server-side we never want the connection to happen.

Reply:
[already replied in person; replying here to document it]

Using `Paths.get()` will work fine so long as the input is `String` not `URI`. If it's `String` it will use the default filesystem provider no matter what the scheme is. I think using `Path` will help us in general so we can avoid code that munges paths directly, and possibly make transition to the cloud-nio library easier.

Reply:
I tried this out, and want to stick with `URI`. `Paths.get` on a `String` doesn't throw an exception, but it also doesn't parse correctly:
```bash
scala> Paths.get("gs://the-bucket/the-file.txt")
res0: java.nio.file.Path = gs:/the-bucket/the-file.txt

scala> res0.toUri
res1: java.net.URI = file:///Users/danmoran/clio/gs:/the-bucket/the-file.txt
```

The "gs://" scheme gets munged into "gs:/". Without a hack to revert this on every `.toString` call, this will break our `gsutil` calls and cause us to write bad values into our JSON blobs.

Reply:
Oh well. OK.